includes/razorpay-subscriptions.php

```diff
@@ -0,0 +1,349 @@
+<?php
+
+require_once __DIR__.'/../razorpay-sdk/Razorpay.php';
+
+use Razorpay\Api\Api;
+
+class RZP_Subscriptions
+{
+    protected $razorpay;
+
+    const RAZORPAY_PLAN_ID               = 'razorpay_wc_plan_id';
+    const INR                            = 'INR';
+
+    public function __construct($keyId, $keySecret)
+    {
+        $this->api = new Api($keyId, $keySecret);
+
+        $this->razorpay = new WC_Razorpay();
+    }
+
+    public function createSubscription($orderId)
+    {
+        global $woocommerce;
+
+        $subscriptionData = $this->getSubscriptionCreateData($orderId);
+
+        try
+        {
+            $subscription = $this->api->subscription->create($subscriptionData);
+        }
+        catch (Exception $e)
+        {
+            echo "Razorpay Error: " . $e->getMessage();
+        }
+
+        // Setting the subscription id as the session variable
+        $sessionKey = $this->getSubscriptionSessionKey($orderId);
+        $woocommerce->session->set($sessionKey, $subscription['id']);
+
+        return $subscription['id'];
+    }
+
+    public function cancelSubscription($subscriptionId)
+    {
+        try
+        {
+            $subscription = $this->api->subscription->cancel($subscriptionId);
+        }
+        catch (Exception $e)
+        {
+            echo "Razorpay Error: " . $e->getMessage();
```

Comment:
@mayankamencherla Can you please add the reason for not rethrowing the exception?